ESLint: Re-enable rule no-access-state-in-setstate

kgabryje opened the pull request:
### SUMMARY
Re-enable ESLint rule `no-access-state-in-setstate`, which was disabled in PR #10839. Code was refactored to fix the errors raised by the rule.

### TEST PLAN
Run `npm run lint`, verify that there are no new Javascript/Typescript errors.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

kgabryje committed:
Re-enable rule no-access-state-in-setstate

mistercrunch commented:
I re-triggered the action in CI, if that doesn't work you can rebase as the issue has been addressed on `master`

kgabryje commented:
@mistercrunch I rebased this PR on the newest master and retriggered the CI a couple of times but it keeps failing the cypress tests (explore/AdhocMetrics.test.ts). @eschutho do you have any idea how we could fix this?

kgabryje committed:
Move accessing event values out of async functions

kgabryje commented:
> @mistercrunch I rebased this PR on the newest master and retriggered the CI a couple of times but it keeps failing the cypress tests (explore/AdhocMetrics.test.ts). @eschutho do you have any idea how we could fix this?

And just as I wrote this comment I figured that the issue is in my code and not in Cypress... I accessed `event.target.value` in an async function and React's good old event pooling made sure that the `target` field was undefined. It's fixed now - I moved accessing the value to synchronous part of event handler. Sorry for confusion!
